pkg/raw/raw_exec.go

```diff
@@ -5,44 +5,98 @@ package raw
 
 import (
 	"fmt"
-	"os/exec"
-	"strconv"
+	"io"
+	"os"
 
+	"github.com/go-logr/logr"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-type Exec struct {
-}
+type Exec struct{}
+
+const filePerm = 0640
 
 func (Exec) Create(filename string, opts ...CreateOption) error {
 	o := &CreateOptions{}
 	o.ApplyOptions(opts)
-	var seek string
-
-	if o.SourceFile == "" && o.Size == nil {
-		return fmt.Errorf("must specify Size when creating without source file")
-	}
+	log := ctrl.Log.WithName("raw-disk")
 
-	ofilename := "of=" + filename
-	if o.Size != nil {
-		seek = "seek=" + strconv.FormatInt(*o.Size, 10) // TODO: verify if the disk size is proper, else G as a suffix
-	}
 	if o.SourceFile == "" {
-		outMsgDD, err := exec.Command("dd", "if=/dev/zero", ofilename, "bs=1", "count=0", seek).Output()
+		if o.Size == nil {
+			return fmt.Errorf("must specify Size when creating without source file")
+		}
+		seek := *o.Size
+		// Position the file cursor one byte before the desired seek position to write a single byte,
+		// to ensure that data is written at the exact byte position specified by seek.
+		err := createEmptyFileWithSeek(log, filename, seek-1)
 		if err != nil {
-			return fmt.Errorf("failed creating the ephemeral disk: %s, exit error %w", string(outMsgDD), err)
+			return fmt.Errorf("failed creating the empty ephemeral disk: %w", err)
```

Comment:
maybe add filename into logs. error on line 75 probably doesn't contain name of file what can be problem

Reply:
Added

Reply:
could you pls do refactor of error messages? Currently filename can be two times in error message. Little depends how error is created. So it doesn't look very consistent.

Reply:
removed the additional filenames from the error messages of copyFile function. createEmptyFileWithSeek function does not contain additional filenames in error messages.